improver_tests/metadata/test_check_datatypes.py

```diff
@@ -178,54 +83,52 @@ def setUp(self):
 
     def test_conformant_cubes(self):
         """Test conformant data, percentile and probability cubes all pass
-        (no error is thrown)"""
+        (no error is thrown and cube is not changed)"""
         cubelist = [
-            self.data_cube, self.probability_cube, self.percentile_cube]
+            self.cube, self.probability_cube, self.percentile_cube]
         for cube in cubelist:
-            check_datatypes(cube)
+            result = cube.copy()
+            check_mandatory_standards(result)
+            self.assertStringEqual(CubeList([cube]).xml(checksum=True),
+                                   CubeList([result]).xml(checksum=True))
 
-    def test_string_coord(self):
-        """Test string coordinate does not throw an error"""
-        self.data_cube.add_aux_coord(
-            AuxCoord(["ukv"], long_name="model", units="no_unit"))
-        check_datatypes(self.data_cube)
+    def test_int64_cube_data(self):
```

Comment:
I think the old code used to check that integer data was np.int32. Is this an intended change to the behaviour?

Reply:
Yes, I think you are right. I have reinstated the requirement for cube.data to be int32 (if an integer type).